webroot/src/components/Licensee/LicenseeSearch/LicenseeSearch.ts

```diff
@@ -0,0 +1,180 @@
+                enforceMax: true,
+            }),
+            ssn: new FormInput({
+                id: 'ssn',
+                name: 'ssn',
+                label: computed(() => this.$t('licensing.ssn')),
+                placeholder: '000-00-0000',
+                validation: Joi.string().pattern(/^([0-9]){3}-([0-9]{2}-([0-9]{4}$))/).allow('').messages({
+                    ...this.joiMessages.string,
+                    'string.pattern.base': this.$t('inputErrors.ssnFormat'),
+                }),
+            }),
+            state: new FormInput({
+                id: 'state',
+                name: 'state',
+                label: computed(() => this.$t('common.state')),
+                valueOptions: this.stateOptions,
+            }),
+            submit: new FormInput({
+                isSubmitInput: true,
+                id: 'submit',
+            }),
+        });
+        this.watchFormInputs(); // Important if you want automated form validation
+    }
+
+    formatSsn(): void {
+        const { ssn } = this.formData;
+        const format = (ssnInputVal) => {
+            // Remove all non-dash and non-numerals
+            let formatted = ssnInputVal.replace(/[^\d-]/g, '');
+
+            // Add the first dash if a number from the second group appears
+            formatted = formatted.replace(/^(\d{3})-?(\d{1,2})/, '$1-$2');
+
+            // Add the second dash if a number from the third group appears
+            formatted = formatted.replace(/^(\d{3})-?(\d{2})-?(\d{1,4})/, '$1-$2-$3');
+
+            // Remove misplaced dashes
+            formatted = formatted.split('').filter((val, idx) => val !== '-' || idx === 3 || idx === 6).join('');
+
+            // Enforce max length
+            return formatted.substring(0, 11);
+        };
+
+        ssn.value = format(ssn.value);
+    }
+
+    async handleSubmit(): Promise<void> {
+        this.validateAll({ asTouched: true });
+        this.customValidateLastName();
+
+        if (this.isFormValid) {
+            this.startFormLoading();
+
+            const allowedSearchProps = [
+                'firstName',
+                'lastName',
+                'ssn',
+                'state'
+            ];
+            const searchProps: LicenseSearch = {};
+
+            allowedSearchProps.forEach((searchProp) => { searchProps[searchProp] = this.formValues[searchProp]; });
+            this.$emit('searchParams', searchProps);
+
+            this.endFormLoading();
+        }
+    }
+
+    customValidateLastName(asTouched = true): void {
+        const { firstName, lastName } = this.formData;
+        const shouldSkip = (asTouched) ? false : !lastName.isTouched;
+
+        if (!shouldSkip && firstName.value && !lastName.value) {
+            lastName.isValid = false;
+            lastName.errorMessage = this.$t('inputErrors.lastNameRequired');
+        } else if (lastName.isValid === false) {
```

Comment:
Can we just do this?

```suggestion
        } else if (!lastName.isValid) {
```

Reply:
Done